src/main/java/net/minecraftforge/event/entity/living/LivingSetAttackTargetEvent.java

```diff
@@ -27,16 +28,48 @@
  **/
 public class LivingSetAttackTargetEvent extends LivingEvent
 {
-
-    private final LivingEntity target;
+    private final ILivingTargetType targetType;
+    private final LivingEntity originalTarget;
+    
     public LivingSetAttackTargetEvent(LivingEntity entity, LivingEntity target)
     {
         super(entity);
-        this.target = target;
+        this.targetType = LivingTargetType.MOB_TARGET;        
+        this.originalTarget = target;
+    }
+    
+    public LivingSetAttackTargetEvent(LivingEntity entity, LivingEntity target, ILivingTargetType targetType)
+    {
+        super(entity);
+        this.targetType = targetType;
+        this.originalTarget = target;
     }
 
+    /**
+     * {@return the target this living entity had after changing its target to a new
+     * target, but before posting this event.}
+     */
     public LivingEntity getTarget()
     {
-        return target;
+        return originalTarget;
+    }
+    
+    /**
+     * {@return the target type of this event.}
+     */
+    public ILivingTargetType getTargetType()
+    {
+        return targetType;
+    }
+    
+    public static interface ILivingTargetType
+    {
+        
+    }
+    
+    public static enum LivingTargetType implements ILivingTargetType
+    {
+        MOB_TARGET,
+        BEHAVIOR_TARGET;
```

Comment:
Documentation on the differences between the two targets.

Reply:
Fixed: c8274622fe30303ec1028dfd70d3841605bb9459